overlord/snapstate/handlers.go

```diff
@@ -318,27 +318,55 @@ func (m *SnapManager) installOneBaseOrRequired(t *state.Task, snapName string, c
 	if err != nil {
 		return nil, err
 	}
-	if isInstalled {
-		return updatePrereqIfOutdated(t, snapName, contentAttrs, userID, flags)
-	}
 
 	deviceCtx, err := DeviceCtx(st, t, nil)
 	if err != nil {
 		return nil, err
 	}
 
-	// in progress?
-	if linkTask, err := findLinkSnapTaskForSnap(st, snapName); err != nil {
-		return nil, err
-	} else if linkTask != nil {
+	inProgress := func(snapName string) (bool, error) {
+		linkTask, err := findLinkSnapTaskForSnap(st, snapName)
```

Comment:
don't we need to change findLinkSnapTaskForSnap itself to account for WaitStatus?

Reply:
The base's link task is never actually in WaitStatus, it's the kernel task that goes into WaitStatus,  so I don't think we need to do that. The base runs up to the auto-connect, then the gadget and then kernel, then the link-snap task of the kernel is put into a WaitStatus and the reboot is requested